src/crypto/crypto.cpp

```diff
@@ -749,4 +752,13 @@ POP_WARNINGS
     sc_sub(&h, &h, &sum);
     return sc_isnonzero(&h) == 0;
   }
+
+  void crypto_ops::derive_view_tag(const key_derivation &derivation, size_t output_index, view_tag &view_tag) {
+    ec_scalar scalar;
+    derivation_to_scalar(derivation, output_index, scalar);
+    const char salt[9] = "view_tag"; // separate domain for view tag
+    uint8_t view_tag_full; // siphash result will be 8 bytes
+    siphash(&salt, sizeof(salt), &scalar, &view_tag_full, 8); // note that siphash will only use the first 16 bytes of the scalar
+    memcpy(&view_tag, &view_tag_full, 1); // only need the first byte to realize optimal perf/space efficiency
+  }
```

Comment:
```suggestion

  void crypto_ops::derive_view_tag(const key_derivation &derivation, size_t output_index, view_tag &view_tag) {
    const char hash[8 + ((sizeof(size_t) * 8 + 6) / 7)];
    uint8_t view_tag_full; // siphash result will be 8 bytes
    char siphash_key[16];
    hash = "view_tag";  // view tag domain-separator
    char* end = hash + 8;
    tools::write_varint(end, output_index);  // add output index
    assert(end <= hash + 8 + sizeof(converted_index));
    for (std::size_t i{0}; i < 16; ++i)
        siphash_key[i] = derivation.data[i];

    // view_tag_full = H[siphash_key](salt, output_index)
    siphash(hash, end - hash, siphash_key, &view_tag_full, 8);

    memcpy(&view_tag, &view_tag_full, 1);  // only need the first byte to realize optimal perf/space efficiency

    memwipe(siphash_key, 16);
  }
```

Only one hash is needed.